plugin/log/log.go

```diff
@@ -26,21 +26,24 @@ type Logger struct {
 // ServeDNS implements the plugin.Handler interface.
 func (l Logger) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
 	state := request.Request{W: w, Req: r}
+	name := state.Name()
 	for _, rule := range l.Rules {
-		if !plugin.Name(rule.NameScope).Matches(state.Name()) {
+		if !plugin.Name(rule.NameScope).Matches(name) {
 			continue
 		}
 
 		rrw := dnstest.NewRecorder(w)
 		rc, err := plugin.NextOrFailure(l.Name(), l.Next, ctx, rrw, r)
 
-		tpe, _ := response.Typify(rrw.Msg, time.Now().UTC())
-		class := response.Classify(tpe)
 		// If we don't set up a class in config, the default "all" will be added
 		// and we shouldn't have an empty rule.Class.
 		_, ok := rule.Class[response.All]
-		_, ok1 := rule.Class[class]
-		if ok || ok1 {
+		if !ok {
```

Comment:
I don't find this use of the variable `ok` readable, the `ok || ok1` might have been slower, but at least it was clearer.
Does a `switch` help here?

Reply:
The variable name isn't great, but the goal here is to skip the call to [`response.Typify()`](https://github.com/charlievieth/coredns/blob/cev/log-perf/plugin/pkg/response/typify.go#L57) if we already know the message needs to be logged: https://github.com/coredns/coredns/blob/72c84dcb1cb7e69941eca7050153e80a5c9f1cf7/plugin/log/log.go#L42

Reply:
Fixed.  I renamed `ok` => `shouldLog`.  Happy to add comments if you think they'll be helpful.